### PR TITLE
Slight HTML Button Improvements

### PR DESCRIPTION
**Pull request by PartMan7**
Changes this makes:
a) Lets people use `/pm` instead of `/msg` in buttons, which is a lot more intuitive.
b) Permits the author of the HTML to direct messages towards themselves via buttons (useful for when Bots create subroomgroupchats and become #, or if they suddenly want hugs from random people).

(Didn't add /w since line limits are worse than horizontally sliced bread)

### CODE REVIEW

**Comment by Zarel** [comment]
Just make an intermediate variable. It's more readable that way, anyway. :|

File: server/chat.ts [hunk 1]
```diff
@@ -1293,17 +1293,17 @@ export class CommandContext extends MessageContext {
 					if ((!this.room || this.room.settings.isPersonal || this.room.settings.isPrivate === true) && !this.user.can('lock')) {
 						const buttonName = / name ?= ?"([^"]*)"/i.exec(tagContent)?.[1];
 						const buttonValue = / value ?= ?"([^"]*)"/i.exec(tagContent)?.[1];
-						if (buttonName === 'send' && buttonValue?.startsWith('/msg ')) {
-							const [pmTarget] = buttonValue.slice(5).split(',');
+						if (buttonName === 'send' && (buttonValue?.startsWith('/msg ') || buttonValue?.startsWith('/pm '))) {
+							const [pmTarget] = buttonValue.replace(/^\/(?:msg|pm) /, '').split(',');
```
Comment (Zarel) [inline server/chat.ts:1297]:
```suggestion
						const msgCommandRegex = /^\/(?:msg|pm|w) /;
						if (buttonName === 'send' && msgCommandRegex.test(buttonValue || '')) {
							const [pmTarget] = buttonValue.replace(msgCommandRegex, '').split(',');
```

**Comment by Zarel** [comment]
Good change, though!

**Comment by PartMan7** [comment]
Come to think of it, this should probably be case-insensitive.

File: server/chat.ts (hunk 1 above)
Comment (PartMan7) [reply server/chat.ts]: Right, I didn't think of that. Thanks. n.n

File: server/chat.ts [hunk 2]
```diff
@@ -1293,17 +1293,18 @@ export class CommandContext extends MessageContext {
 					if ((!this.room || this.room.settings.isPersonal || this.room.settings.isPrivate === true) && !this.user.can('lock')) {
 						const buttonName = / name ?= ?"([^"]*)"/i.exec(tagContent)?.[1];
 						const buttonValue = / value ?= ?"([^"]*)"/i.exec(tagContent)?.[1];
-						if (buttonName === 'send' && buttonValue?.startsWith('/msg ')) {
-							const [pmTarget] = buttonValue.slice(5).split(',');
+						const msgCommandRegex = /^\/(?:msg|pm|w|whisper) /i;
+						if (buttonName === 'send' && msgCommandRegex.test(buttonValue || '')) {
+							const [pmTarget] = buttonValue.replace(msgCommandRegex, '').split(',');
```
Comment (mia-pi-git) [inline server/chat.ts:1298]:
```suggestion
						const msgCommandRegex = /^\/(?:msg|pm|w|whisper) /i;
						if (buttonName === 'send' && buttonValue && msgCommandRegex.test(buttonValue)) {
							const [pmTarget] = buttonValue.replace(msgCommandRegex, '').split(',');
```

Not sure why it's so low, but your check failure is coming from here. RegExp.exec can return null, so you need to assert that it exists - the ``|| ''`` isn't what you want to do there.

Comment (Zarel) [reply server/chat.ts:1298]: Yeah, this was my mistake thinking through the logic wrong.

**Pull request merged**